ui/desktop/src/components/pair.tsx

```diff
@@ -80,6 +80,20 @@ export default function Pair({
   // Handle initial message from hub page
   useEffect(() => {
     const messageFromHub = location.state?.initialMessage;
+    const resetChat = location.state?.resetChat;
+
+    // If we have a resetChat flag from Hub, clear any existing recipe config
+    if (resetChat) {
+      const newChat: ChatType = {
+        ...chat,
+        recipeConfig: null,
+        recipeParameters: null,
+        title: 'New Chat',
+        messages: [], // Clear messages for fresh start
+        messageHistoryIndex: 0,
+      };
```

Comment:
this should probably be/use a constant seeing that we have 'New Chat' already 4 times